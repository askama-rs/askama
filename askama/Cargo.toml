[package]
name = "askama"
version = "0.13.0"
description = "Type-safe, compiled Jinja-like templates for Rust"
keywords = ["markup", "template", "jinja2", "html"]
categories = ["template-engine"]
homepage = "https://askama.readthedocs.io/"
repository = "https://github.com/askama-rs/askama"
license = "MIT OR Apache-2.0"
edition = "2021"
rust-version = "1.81"

[package.metadata.docs.rs]
features = ["full"]
rustdoc-args = ["--generate-link-to-definition", "--cfg=docsrs"]

[[bench]]
name = "to-json"
harness = false
required-features = ["serde_json"]

[[bench]]
name = "escape"
harness = false

[dependencies]
itoa = "1.0.11"

# needed by feature "derive"
askama_derive = { version = "=0.13.0", path = "../askama_derive", default-features = false, optional = true }

# needed by feature "serde_json"
serde = { version = "1.0", optional = true, default-features = false }
serde_json = { version = "1.0", optional = true, default-features = false }

# needed by feature "urlencode"
percent-encoding = { version = "2.1.0", optional = true, default-features = false }

[dev-dependencies]
assert_matches = "1.5.0"
criterion = "0.5"

[badges]
maintenance = { status = "actively-developed" }

[features]
default = ["config", "derive", "std", "urlencode", "askama_derive?/default"]
full = ["default", "blocks", "code-in-doc", "serde_json", "askama_derive?/full"]

alloc = [
    "askama_derive?/alloc",
    "serde?/alloc",
    "serde_json?/alloc",
    "percent-encoding?/alloc",
]
blocks = ["askama_derive?/blocks"]
code-in-doc = ["askama_derive?/code-in-doc"]
config = ["askama_derive?/config"]
derive = ["askama_derive"]
serde_json = ["std", "askama_derive?/serde_json", "dep:serde", "dep:serde_json"]
std = [
    "alloc",
    "askama_derive?/std",
    "serde?/std",
    "serde_json?/std",
    "percent-encoding?/std",
]
urlencode = ["askama_derive?/urlencode", "dep:percent-encoding"]
