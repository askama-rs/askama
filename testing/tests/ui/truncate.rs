use askama::Template;

#[derive(Template)]
#[template(source = r#"{{ text | truncate }}"#, ext = "html")]
struct NoArgument<'a> {
    text: &'a str,
}

#[derive(Template)]
#[template(source = r#"{{ text | truncate(length) }}"#, ext = "html")]
struct WrongArgumentType<'a> {
    text: &'a str,
    length: f32,
}

#[derive(Template)]
#[template(source = r#"{{ text | truncate(length, extra) }}"#, ext = "html")]
struct TooManyArguments<'a> {
    text: &'a str,
    length: usize,
    extra: bool,
}

fn main() {}
