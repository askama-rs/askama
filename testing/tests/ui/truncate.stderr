error: `length` argument is missing when calling `truncate` filter; its arguments are: (length)
 --> NoArgument.html:1:3
       "text | truncate }}"
 --> tests/ui/truncate.rs:4:21
  |
4 | #[template(source = r#"{{ text | truncate }}"#, ext = "html")]
  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `truncate` filter accepts at most 1 argument; its arguments are: (length)
 --> TooManyArguments.html:1:27
       "extra) }}"
  --> tests/ui/truncate.rs:17:21
   |
17 | #[template(source = r#"{{ text | truncate(length, extra) }}"#, ext = "html")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0277]: the trait bound `usize: TryFrom<f32>` is not satisfied
 --> tests/ui/truncate.rs:9:10
  |
9 | #[derive(Template)]
  |          ^^^^^^^^ the trait `From<f32>` is not implemented for `usize`
  |
  = help: the following other types implement trait `From<T>`:
            `usize` implements `From<bool>`
            `usize` implements `From<std::ptr::Alignment>`
            `usize` implements `From<u16>`
            `usize` implements `From<u8>`
  = note: required for `f32` to implement `Into<usize>`
  = note: required for `usize` to implement `TryFrom<f32>`
  = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)
